lib/StylelintPluginError.js

```diff
@@ -0,0 +1,17 @@
+'use strict';
+
+let Base;
+
+try {
+  // this wasn't added to webpack until v2.5.0, and peerDeps state this package
+  // supports webpack@1.x.
+  Base = require('webpack/lib/WebpackError');
+} catch (e) {
+  Base = Error;
+}
+
+module.exports = class StylelintPluginError extends Base {
+  constructor(message) {
+    super(message);
```

Comment:
TIL that ES2015 will automatically do these kinds of constructors, so this export could just be:

```js
module.exports = class StylelintPluginError extends Base {};
```

and can become more complicated later on as necessary.

Reply:
I prefer being verbose here. It's especially relevant when the RFC changes are implemented, as it'll likely need to do some things with the constructor.